Source/WebCore/PAL/pal/graphics/WebGPU/Impl/WebGPUDeviceImpl.cpp

```diff
@@ -175,9 +175,23 @@ Ref<BindGroupLayout> DeviceImpl::createBindGroupLayout(const BindGroupLayoutDesc
 {
     auto label = descriptor.label.utf8();
 
-    auto backingEntries = descriptor.entries.map([this] (const auto& entry) {
-        return WGPUBindGroupLayoutEntry {
-            nullptr,
+    auto backingExternalTextureEntries = descriptor.entries.map([] (const auto&) {
```

Comment:
It's not really necessary to iterate over descriptors twice right?

Previously, there was a single call to `map`. This change keeps the call to `map` and also iterates over the data a second time.

It seems like it would be a simpler change to just store the chained entries into a vector if an external entry is encountered.

Reply:
I think it has to be 2 different iterations, because there are pointers from `backingEntries` into `backingExternalTextureEntries`. So, if there are any modifications to `backingExternalTextureEntries`, the vector's backing store could be reallocated, and the pointers will be dangling. So I don't think we can build them in tandem; I think we need to build the 2 vectors one after the other.

Reply:
Oh great point, actually we have that problem in `Ref<BindGroup> DeviceImpl::createBindGroup(const BindGroupDescriptor& descriptor)` for which I just filed https://bugs.webkit.org/show_bug.cgi?id=257192

I suppose a container which does not re-allocate the underlying storage could be used given the size of this container is often 0 or 1, but using Vector seems fine here

Reply:
Yep. I'll make a follow-up patch to migrate all the places where we use this pattern to SegmentedVector.